File: easy_NumberAddition.py
"""
Traverse a string,
add up all the numbers
and return the final result

ex: '55blah 4' = 55 + 4 = 59
"""
def NumberAddition(str): 
  sum = 0
  temp = '0'
  for x in str:
    if x.isdigit():
      temp += x
    else:
      sum += int(temp)
      temp = '0'
  return sum + int(temp)
    
print NumberAddition(raw_input())           

File: easy_NumberAddition.java
import java.util.Scanner;

public class Easy_NumberAddition {

	public static int function(String input){
		char[] charInput = input.toCharArray(); //make string into char array
		String temp = "0";
		int result = 0;
		for(char x : charInput){
			if(Character.isDigit(x)){
				temp += x; //if character is digit save it
			} else {
				result += Integer.parseInt(temp);
				temp = "0"; //if it's not a digit, add saved digits to result
			}
		}
		if(temp != ""){
			result += Integer.parseInt(temp); //the last character could be a digit
		}
		return result;
	}
	
	public static void main(String[] args) {
		System.out.println( function( new Scanner(System.in).nextLine() ) );
	}
}
